### PR TITLE
[linux] initial port and osal additions.

### PR DESCRIPTION
**Pull request by turon**
- Created linux port of os_task, os_sem, os_mutex, os_callout, os_eventq, etc.
- Added unit tests for osal.
- Mods to support pointer alignment on 64-bit architectures.
- Fixes to ble_hci_sock to support linux port
- Fallback to BLE 4.2 or 4.0 controllers when 5.0 features not available.

Tested advertising, scan response, connections and GATT database operations.

### CODE REVIEW

File: porting/common/src/sysinit/sysinit.c [hunk 1]
```diff
@@ -34,7 +36,8 @@ sysinit_dflt_panic_cb(const char *file, int line, const char *func,
     }
 #endif
 
-    __assert_func(file, line, func, expr);
+    __assert_fail(msg, file, line, func);
+    //__assert_func(file, line, func, expr);
```
Comment (turon) [inline porting/common/src/sysinit/sysinit.c]: This is a known conflict between FreeRTOS and Linux that needs to be resolved before merging.

**Comment by sterlinghughes** [comment]
@andrzej-kaczmarek are we good to merge this?

**Comment by andrzej-kaczmarek** [comment]
@sterlinghughes I need to update porting layer and push it to latest official master - I'm on it. Then this can be put on top, perhaps after some changes.

File: porting/linux/os/os_atomic.c [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <stdint.h>
+#include <pthread.h>
+
+pthread_mutex_t s_mutex = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
+pthread_cond_t s_cond = PTHREAD_COND_INITIALIZER;
```
Comment (sjanc) [inline porting/linux/os/os_atomic.c]: should be static?

File: porting/linux/config/sdk_config.h [hunk 3]
```diff
@@ -0,0 +1,698 @@
+
```
[the remark this answers is not in the document]
Comment (sjanc) [reply porting/linux/config/sdk_config.h]: is this needed anyway for host?

**Comment by turon** [comment]
Usurped by #90.

**Pull request closed without merging**